api/v1alpha1/virtualmachine_conversion.go

```diff
@@ -1121,6 +1152,22 @@ func Convert_v1alpha1_VirtualMachine_To_v1alpha3_VirtualMachine(in *VirtualMachi
 		}
 	}
 
+	// For existing v1a1 VMs that have metadata transport resource type set to use
+	// a configMap, set an annotation during up-conversion to indicate that the
+	// provider needs to also use the configMap. The annotation will also be
+	// used to set the respective ConfigMapName field on down-conversion to v1alpha1
+	md := in.Spec.VmMetadata
+	if md != nil && !apiequality.Semantic.DeepEqual(*md, VirtualMachineMetadata{}) && md.ConfigMapName != "" {
```

Comment:
I think we can get rid of the `DeepEqual()` because `ConfigMapName` wouldn't be empty